tests/local/network.go

```diff
@@ -300,53 +336,66 @@ func (n *localNetwork) deployTeleporterRegistryContracts(
 		},
 	}
 
-	for _, subnetInfo := range n.GetSubnetsInfo() {
+	subnets := n.getAllSubnetsInfo()
+	for _, subnetInfo := range subnets {
 		opts, err := bind.NewKeyedTransactorWithChainID(deployerKey, subnetInfo.EVMChainID)
 		Expect(err).Should(BeNil())
 		teleporterRegistryAddress, tx, _, err := teleporterregistry.DeployTeleporterRegistry(
 			opts, subnetInfo.RPCClient, entries,
 		)
 		Expect(err).Should(BeNil())
-
-		n.subnetsInfo[subnetInfo.SubnetID].TeleporterRegistryAddress = teleporterRegistryAddress
 		// Wait for the transaction to be mined
 		utils.WaitForTransactionSuccess(ctx, subnetInfo, tx)
 
+		if subnetInfo.SubnetID == constants.PrimaryNetworkID {
+			n.primaryNetworkInfo.TeleporterRegistryAddress = teleporterRegistryAddress
+		} else {
+			n.subnetsInfo[subnetInfo.SubnetID].TeleporterRegistryAddress = teleporterRegistryAddress
+		}
+
 		log.Info("Deployed TeleporterRegistry contract to subnet", subnetInfo.SubnetID.Hex(),
 			"Deploy address", teleporterRegistryAddress.Hex())
 	}
 
 	log.Info("Deployed TeleporterRegistry contracts to all subnets")
 }
 
-func (n *localNetwork) GetSubnetsInfo() []interfaces.SubnetTestInfo {
+func (n *LocalNetwork) GetSubnetsInfo() []interfaces.SubnetTestInfo {
 	return []interfaces.SubnetTestInfo{
 		*n.subnetsInfo[n.subnetAID],
 		*n.subnetsInfo[n.subnetBID],
-		*n.subnetsInfo[n.subnetCID],
 	}
 }
 
-func (n *localNetwork) GetTeleporterContractAddress() common.Address {
+func (n *LocalNetwork) GetPrimaryNetworkInfo() interfaces.SubnetTestInfo {
+	return *n.primaryNetworkInfo
+}
+
+func (n *LocalNetwork) getAllSubnetsInfo() []interfaces.SubnetTestInfo {
```

Comment:
can you add a comment that this includes primary network that is appended at the end